File: MojPrijevozFront/moj_prijevoz/lib/resources/responses/search_fare/search_fare_driver_response.dart
// ignore_for_file: public_member_api_docs, sort_constructors_first
import 'package:json_annotation/json_annotation.dart';

import 'package:moj_prijevoz/utils/json_parser.dart';

part 'search_fare_driver_response.g.dart';

@JsonSerializable()
class SearchFareDriverResponse extends JsonResponse {
  @override
  final int id;
  double price;
  double? additionalPrice;
  int userVehicleId;
  int vehicleId;
  SearchFareDriverResponse({
    required this.id,
    required this.price,
    this.additionalPrice,
    required this.vehicleId,
    required this.userVehicleId,
  });

  @override
  Map<String, dynamic> toJson() => _$SearchFareDriverResponseToJson(this);

  factory SearchFareDriverResponse.fromJson(Map<String, dynamic> json) =>
      _$SearchFareDriverResponseFromJson(json);
}

File: MojPrijevoz/MojPrijevoz.Model/Responses/SearchFare/SearchFareDriverResponse.cs
namespace MojPrijevoz.Model.Responses.SearchFare;

public class SearchFareDriverResponse
{
    public int Id { get; set; }
    public int VehicleId { get; set; }
    public int UserVehicleId { get; set; }
    public double Price { get; set; }
    public double? AdditionalPrice { get; set; }
}